internal/models/pubkey_model.go

```diff
@@ -17,3 +28,39 @@ type Pubkey struct {
 	// Public key SHA256 fingerprint. Generated (read-only).
 	Fingerprint string `db:"fingerprint" json:"fingerprint"`
 }
+
+// FingerprintAWS calculates fingerprint used by AWS.
+// AWS calculates MD5 sums differently then all the other tools and use format DER as the base for the sub.
+// see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/verify-keys.html for more details.
+func (pk *Pubkey) FingerprintAWS() (string, error) {
+	pkey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(pk.Body))
+	if err != nil {
+		return "", fmt.Errorf("failed to calculate fingerprint for %s: %w", pk.Name, err)
+	}
+	// when rsa key
+	if pkey.Type() == "ssh-rsa" {
+		return fingerprintRsaAWS(pkey)
+	}
+	// when ed25519 key
+	// this is the same as what we store, but better be sure here
+	return strings.TrimLeft(ssh.FingerprintSHA256(pkey), "SHA256:") + "=", nil
```

Comment:
Are sou sure the padding character is always one? I mean, the hash should be constant length, I am trying to think about situation where it could actually be shorter. I am just thinking loud, this smells.

Reply:
Yeah, I agree proper calculation would be better here, I THINK it is always one from what I've read and calculated on paper, but I agree that proper calculation in code would be more appropriate here.